sycli/src/main.rs

```diff
@@ -665,3 +667,120 @@ fn main() {
         _ => {}
     }
 }
+
+/// Parse search criteria out of a filter string
+fn parse_filter(searches: &str) -> Vec<Criterion> {
+    use rpc::criterion::{Operation, Value};
+
+    let mut criterion = Vec::new();
+
+    // possible delimiters
+    let delimiters = vec![">=", "<=", "==", "!=", "<", ">", ":"];
+
+    'search: for search in searches.split_whitespace() {
+        for delim in &delimiters {
+            if search.split(delim).count() == 2 {
+                let field = search.split(delim).nth(0).unwrap().to_string();
+                let value = search.split(delim).nth(1).unwrap();
+
+                let value = if field == "name"
+                    || field == "path"
+                    || field == "status"
+                    || field == "tracker"
+                {
+                    // name, path, status, and tracker fields are the only kind that search on
+                    // strings
+                    Value::S(value.to_string())
+                } else {
+                    // floating point seems to work fine even for strictly integer types
+                    Value::F(value.parse().expect("Invalid numeric value"))
+                };
+
+                // the delimiter symbol determines the operation type
+                let op = match *delim {
+                    ">=" => Operation::GTE,
+                    "<=" => Operation::LTE,
+                    "==" => Operation::Eq,
+                    "!=" => Operation::Neq,
+                    ">" => Operation::GT,
+                    "<" => Operation::LT,
+                    ":" => Operation::ILike,
+                    _ => unreachable!(),
+                };
+
+                criterion.push(Criterion { field, op, value });
+                continue 'search;
+            }
+        }
+    }
+
+    // if no delimiters found, assume a simple name query
+    if criterion.is_empty() {
```

Comment:
Can you add a test which has a valid filter condition, then invalid, e.g. "name:foo key~val" to show this branch doesn't happen per loop iteration.

Reply:
Also can you add a test which shows that a filter like "spaced name" gets translated into this condition with its whitespace